### PR TITLE
Clara Andrew: back-end-project-week

### PR DESCRIPTION
**Pull request by clarakosi**
## Trello Set Up:
* [x] Create a Trello account
* [x] Create a new board called "Lambda Notes(Backend) - {Your Name}"
* [x] Create lists titled `backlog`,`To Do`, `In Progress`, and `Done`
* [x] Fill in the `To Do` list with the MVP features listed below
* [x] Fill in the `backlog` list with all the extra features listed below
* [x] Share your board with the project manager that has been assigned to you.  If you have not been assigned yet, reach out to your lead TA for guidance
* [x] Add your Trello URL to your project's README.md file.  Commit the change, push it to your repository & submit a pull request

## Backend MVP Features:
* [x] Create a Database for your app
* [x] Create a Node app and connect it to your database
* [x] Create a Registration Page that allows users to create accounts for your app and sign in with email/password.
* [x] Store these User Accounts in your Database
* [x] Add a Sign In Page
* [x] Store Notes in your Database
* [x] Deploy your application to the web

Build an API Endpoint in your Node App for each of the following, and connect them to your React App. These should all be read from your Database and scoped to the logged in user.
* [x] Display a list of notes
* [x] Create a note with a title and content
* [x] View an existing note
* [x] Edit an existing note
* [x] Delete an existing note

### CODE REVIEW

File: client/src/Component/Notes.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+          {filter.map((note, index) => {
+            return (
+              <li key={note._id} className="note" onClick={() => {
+                  this.props.previewNote(note.title, note.content, note._id);
+                }}
+                onMouseOver={this.hoverToggle}
+                onMouseOut={this.mouseOut}
+              >
+                <div className="note--title">
+                  {note.title.length > 30
```
Comment (rholdy) [inline client/src/Component/Notes.js]: Nice! Proactively cleaning up display. Love this.

File: client/src/Component/ViewNotes.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      content: this.state.content
+    };
+    const note = {
+      title: this.state.title,
+      content: this.state.content
+    };
+
+    if (this.state.content.length === 0 || this.state.title.length === 0) {
+      alert(
+        "You need to have a title and Body in your note before you can save"
```
Comment (rholdy) [inline client/src/Component/ViewNotes.js]: Adding validations is often an afterthought. That you've gone to this level of detail is really impressive to me. Well done.

File: client/src/less/homepage.less [hunk 3]
```diff
@@ -0,0 +1,74 @@
+html,
+body {
+  background: @secondary-color;
```
Comment (rholdy) [inline client/src/less/homepage.less]: Less. Nice.

File: controllers/AuthControllers.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
+const User = require("../models/User");
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const saltRounds = 11;
+const secret = process.env.SECRET || "Four for you Glen Coco, you go Glen Coco! And none for Gretchen Weiners. Bye.";
```
Comment (rholdy) [inline controllers/AuthControllers.js]: I'm not sure if this line is boiler plate or not, but if you added this. You get a million gold stars. My dog is named "Glen Coco".

**Comment by rholdy** [comment]
I love this. Lets see you add some tests :)

File: controllers/AuthControllers.js (hunk 4 above)
Comment (clarakosi) [reply controllers/AuthControllers.js]: Not boilerplate but Mean Girls quote =)

**Comment by camilamdaniels** [comment]
Great job! Your code is very well documented and clean. Kudos for finishing early and extending the MVP!

**Pull request closed without merging**